website/client/js/services/userServices.js

```diff
@@ -319,7 +319,9 @@ angular.module('habitrpg')
 
         addTag: function(data) {
           user.ops.addTag(data);
-          Tags.createTag(data.body);
+          Tags.createTag(data.body).then(function(res) {
+            sync();
```

Comment:
All you'd need to do here to return the newly created tag is:

```js
return Tags.createTag(data.body).then(function(res) {
  return res.data.data;
});
```

That will make the `User.addTag` function a promise and you could use that to apply the id to the corresponding tag on the webpage.

I _think_ something like this would work.

```js
User.addTag(data).then(function (tag) {
  data.id = tag.id;
});
```